xm-commons-domainevent-db/src/main/java/com/icthh/xm/commons/domainevent/db/service/kafka/SystemQueueConsumer.java

```diff
@@ -0,0 +1,101 @@
+package com.icthh.xm.commons.domainevent.db.service.kafka;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.icthh.xm.commons.logging.util.MdcUtils;
+import com.icthh.xm.commons.messaging.event.system.SystemEvent;
+import com.icthh.xm.commons.security.XmAuthenticationContextHolder;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import com.icthh.xm.commons.tenant.TenantContextUtils;
+import com.icthh.xm.lep.api.LepManager;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+
+import static com.icthh.xm.commons.lep.XmLepConstants.THREAD_CONTEXT_KEY_AUTH_CONTEXT;
+import static com.icthh.xm.commons.lep.XmLepConstants.THREAD_CONTEXT_KEY_TENANT_CONTEXT;
+
+@Slf4j
+@Service
+public class SystemQueueConsumer {
+
+    private final TenantContextHolder tenantContextHolder;
+    private final XmAuthenticationContextHolder authContextHolder;
+    private final SystemConsumerService systemConsumerService;
+    private final LepManager lepManager;
+
+    public SystemQueueConsumer(TenantContextHolder tenantContextHolder,
+                               XmAuthenticationContextHolder authContextHolder,
+                               SystemConsumerService systemConsumerService,
+                               LepManager lepManager) {
+        this.tenantContextHolder = tenantContextHolder;
+        this.authContextHolder = authContextHolder;
+        this.systemConsumerService = systemConsumerService;
+        this.lepManager = lepManager;
+    }
+
+    /**
+     * Consume system event message.
+     *
+     * @param message the system event message
+     */
+    @Retryable(maxAttemptsExpression = "${application.retry.max-attempts}",
+        backoff = @Backoff(delayExpression = "${application.retry.delay}",
+            multiplierExpression = "${application.retry.multiplier}"))
```

Comment:
how about migrate to Dynamic topic? that support dynamic queue configuration and deadletter?

Reply:
Yes, it's possible.
But I wanted to hear your vision of creation